### PR TITLE
support generate golang dependency

### PR DESCRIPTION
**Pull request by aiwantaozi**

### CODE REVIEW

File: syft/pkg/cataloger/golang/scaffolding_parser.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+func (p *scaffoldingParser) parse() ([]*pkg.Package, []artifact.Relationship, error) {
+	// lock file isn't existed
+	if _, err := os.Stat(filepath.Join(filepath.Dir(p.currentFilepath), GoSumFileName)); err != nil {
+		return parseGoMod(p.currentFilepath, p.reader)
+	}
+
+	log.Infof("parsing golang dependency file %s by scaffolding", p.currentFilepath)
+	var ctx, cancel = context.WithTimeout(context.Background(), 5*time.Minute)
```
Comment (gitlawr) [inline syft/pkg/cataloger/golang/scaffolding_parser.go:58]: Why do we need the timeout?

File: syft/pkg/cataloger/golang/scaffolding_parser.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+func (p *scaffoldingParser) parseRootModule(ctx context.Context) (*pkg.Package, error) {
+	buf := new(bytes.Buffer)
+	defer buf.Reset()
+
+	cmd, err := p.getRootModCommand(ctx, buf)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = cmd.Run(); err != nil {
```
Comment (gitlawr) [inline syft/pkg/cataloger/golang/scaffolding_parser.go]: Should we handle stderr? This is usually the ExitError that does not provide sufficient info

File: syft/pkg/cataloger/golang/scaffolding_parser.go (hunk 1 above)
Comment (aiwantaozi) [reply syft/pkg/cataloger/golang/scaffolding_parser.go:58]: Just prevent command runs for a long time. Even if we have not encountered the situation of being stuck now.

File: syft/pkg/cataloger/golang/scaffolding_parser.go (hunk 2 above)
Comment (gitlawr) [reply syft/pkg/cataloger/golang/scaffolding_parser.go]: I mean cmd.Stderr. Printing the command does not help gathering the error message

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by thxCode — APPROVED** [review]
LGTM

**Pull request merged**